internal/pokt_client_decorators/cached_client_test.go

```diff
@@ -0,0 +1,204 @@
+
+				suite.newCachedClient.lastFailure = time.Now()
+
+				suite.mockTTLCachedService.EXPECT().Get("test-test").Return(ttlcacheItem)
+
+			},
+			expectedResponse: nil,
+			expectedError:    ErrRecentlyFailed,
+		},
+		{
+			name: "Cached",
+			setupMocks: func() {
+
+				suite.mockTTLCachedService.EXPECT().Get("test-test").Return(ttlcacheItem)
+
+				suite.mockPocketService.EXPECT().GetSession(testRequest).Return(testResponse, nil)
+
+				suite.mockTTLCachedService.EXPECT().Set("test-test", testResponse, ttlcache.DefaultTTL).Return(ttlcacheItem)
+
+			},
+			expectedResponse: testResponse,
+			expectedError:    nil,
+		},
+		{
+			name: "Error",
+			setupMocks: func() {
+
+				suite.mockTTLCachedService.EXPECT().Get("test-test").Return(ttlcacheItem)
+
+				suite.mockPocketService.EXPECT().GetSession(testRequest).Return(nil, errUnderlayingProvider)
+
+			},
+			expectedResponse: nil,
+			expectedError:    errUnderlayingProvider,
+		},
+	}
+
+	// run test cases
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+
+			suite.SetupTest() // reset mocks
+
+			tc.setupMocks() // setup mocks
+
+			session, err := suite.newCachedClient.GetSession(testRequest)
+
+			// assert results
+			suite.Equal(tc.expectedResponse, session)
+			suite.Equal(tc.expectedError, err)
+
+		})
+	}
+
+}
+
+// test SendRelay using table driven tests
+func (suite *CachedClientTestSuite) TestSendRelay() {
+
+	testSendRelayResponse := &models.SendRelayResponse{
+		Response: "test",
+	}
+
+	// create test cases
+	testCases := []struct {
+		name             string
+		request          *models.SendRelayRequest
+		setupMocks       func(*models.SendRelayRequest)
+		expectedResponse *models.SendRelayResponse
+		expectedError    error
+	}{
+		{
+			name: "InvalidRequest",
+			request: &models.SendRelayRequest{
+				Payload:            nil, // invalid request
+				Signer:             nil, // invalid request
+				Chain:              "test",
+				SelectedNodePubKey: "test",
+				Session:            &models.Session{},
+			},
+			setupMocks: func(request *models.SendRelayRequest) {
+
+				suite.mockPocketService.EXPECT().SendRelay(request).Return(nil, models.ErrMalformedSendRelayRequest)
+
+			},
+			expectedResponse: nil,
+			expectedError:    models.ErrMalformedSendRelayRequest,
+		},
+		{
+			name: "SessionError",
+			request: &models.SendRelayRequest{
+				Payload: &models.Payload{},
+				Signer: &models.Ed25519Account{
+					PublicKey: "test",
+				},
+				Chain:              "test",
+				SelectedNodePubKey: "test",
+			},
+			setupMocks: func(request *models.SendRelayRequest) {
+
+				suite.mockTTLCachedService.EXPECT().Get("test-test").Return(&ttlcache.Item[string, *models.GetSessionResponse]{})
+
+				suite.mockPocketService.EXPECT().GetSession(&models.GetSessionRequest{
+					AppPubKey: "test",
+					Chain:     "test",
+				}).Return(nil, errors.New("error"))
+
+			},
+			expectedResponse: nil,
+			expectedError:    errors.New("error"),
+		},
+		{
+			name: "Success",
```

Comment:
I like this test case, but we should have a test case where it calls underlying session provider and it returns a valid session as well.